reeds/function_libs/optimization/eds_energy_offsets.py

```diff
@@ -94,10 +94,11 @@ def estimate_energy_offsets(ene_trajs: List[pd.DataFrame], initial_offsets: List
     else: 
         undersampling_eoffs = all_eoffs[undersampling_idx:]
         results = 'Undersampling found at replica ' + str(undersampling_idx+1) \
-                  + ' with s = ' + str(s_values[undersampling_idx]) + '\n\n' 
+                  + ' with s = ' + str(s_values[undersampling_idx-1]) + '\n\n' 
```

Comment:
is this correct? It's a bit counter-intuitive, that the undersampling index is `undersampling_idx+1` but the location in the array is `undersampling_idx-1`

Reply:
I remember having to change something here, but really can't remember the details. 
Maybe it is just a typo an they should both be undersampling_idx+1?

Reply:
I think it needs to be a +1

Reply:
sorry, it must be a minus for both

here is the output with two +1 s
'''
Fractions of undersampling occurrence sampling
fraction treshold: 0.9

| s             |       state 1 |state 2        |state 3        |state 4        |state 5|
|---    |---    |---    |---    |---    |---    |
| 1.00000       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.73600       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.54100       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.39800       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.29300       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.21500       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.15900       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.11700       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.08580       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.06310       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.04640       |       0.00    |0.00   |0.00   |0.47   |0.54   |
| 0.03410       |       0.00    |0.00   |0.00   |0.49   |0.57   |
| 0.02510       |       0.03    |0.01   |0.00   |0.99   |0.99   |
| 0.01850       |       0.88    |0.70   |0.00   |1.00   |1.00   |
| 0.01360       |       0.99    |0.97   |0.01   |1.00   |1.00   |
| 0.01100       |       0.98    |0.97   |0.85   |1.00   |1.00   |
| 0.01000       |       0.98    |0.97   |0.85   |1.00   |1.00   |
| 0.00870       |       0.98    |0.98   |0.97   |0.99   |1.00   |
| 0.00750       |       0.99    |0.98   |0.98   |0.99   |1.00   |
| 0.00650       |       0.99    |0.99   |0.99   |0.99   |1.00   |
| 0.00560       |       0.99    |0.99   |0.99   |1.00   |1.00   |

Undersampling found at replica 19 with s = 0.0065
'''

here with -1s:
'''
Fractions of undersampling occurrence sampling
fraction treshold: 0.9

| s             |       state 1 |state 2        |state 3        |state 4        |state 5|
|---    |---    |---    |---    |---    |---    |
| 1.00000       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.73600       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.54100       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.39800       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.29300       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.21500       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.15900       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.11700       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.08580       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.06310       |       0.00    |0.00   |0.00   |0.47   |0.53   |
| 0.04640       |       0.00    |0.00   |0.00   |0.47   |0.54   |
| 0.03410       |       0.00    |0.00   |0.00   |0.49   |0.57   |
| 0.02510       |       0.03    |0.01   |0.00   |0.99   |0.99   |
| 0.01850       |       0.88    |0.70   |0.00   |1.00   |1.00   |
| 0.01360       |       0.99    |0.97   |0.01   |1.00   |1.00   |
| 0.01100       |       0.98    |0.97   |0.85   |1.00   |1.00   |
| 0.01000       |       0.98    |0.97   |0.85   |1.00   |1.00   |
| 0.00870       |       0.98    |0.98   |0.97   |0.99   |1.00   |
| 0.00750       |       0.99    |0.98   |0.98   |0.99   |1.00   |
| 0.00650       |       0.99    |0.99   |0.99   |0.99   |1.00   |
| 0.00560       |       0.99    |0.99   |0.99   |1.00   |1.00   |

Undersampling found at replica 17 with s = 0.0087
'''

Reply:
yes, that looks good! thanks for the fix :)

Reply:
@SchroederB If I am counting right, the replica with s = 0.0087 is replica number 18 and not 17. Unless you count the first replica (s = 1) as replica 0. 

I was always counting it with the first element being 1 and not 0, but maybe you where thinking otherwise? (starting with 1 makes it consistent with most plots which have a filename ending in _X where X is the replica number.)

Reply:
Yes, for me it made sense to add 1 to the replica index when printing it because the replica numbering starts at 1, as Candide said. But then of course, not add +1 when referring to the corresponding s value:
 
results = 'Undersampling found at replica ' + str(undersampling_idx+1) \
                  + ' with s = ' + str(s_values[undersampling_idx]) + '\n\n'

Reply:
aaaaaaaaaah I see!
hmmm.... I was thinking with a 0.
But I see your point with the 1. Hmm ok, but then it is not a -1  for the first value and  we keep the -1. 
To be honest I don't have hard feelings on starting with 0 or 1.

Reply:
I changed it :)